### PR TITLE
AC-665 Updated UI for Login Page

### PR DESCRIPTION
**Pull request by rishabh-997**
## Description of what I changed
1. Created a better UI design
2. Used constraint layout to flatten the hierarchy as much as possible
3. Used View Binding instead of binding views explicitly so that changing view type won't affect code.
4. Refactored code to reduce ~100 lines of code in LoginFragment

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-665

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by rishabh-997** [comment]
WhatsApp Image 2020-05-18 at 4 20 27 PM

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginPresenter.java [hunk 1]
```diff
@@ -235,7 +237,7 @@ public void onResponse(@NonNull Call<Results<Location>> call, @NonNull Response<
                         loginView.startFormListService();
                         loginView.setLocationErrorOccurred(false);
                     } else {
-                        loginView.showInvalidURLSnackbar("Failed to fetch server's locations");
+                        loginView.showInvalidURLSnackbar("The entered server is currently down, please select a different one.");
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginPresenter.java:240]: Put this string as a string resource instead

Comment (f4ww4z) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginPresenter.java:240]: Use `snackbar_server_error`

**Comment by rishabh-997** [comment]
I tried it and it wasn't looking appealing

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginPresenter.java (hunk 1 above)
Comment (rishabh-997) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginPresenter.java]: Done,
1. Had to pass context from activity to extract string resource from not Activity class.
2. Had to update tests accordingly.
3. Uploaded screenshots with the `Get Started` aligned both ways. It is currently aligned to left in the PR, so if you like the central way, I'll update the PR

**Comment by VibhorChinda** [comment]
> I tried it and it wasn't looking appealing. I can add another screenshot here with that text vertically alligned in middle

@rishabh-997 if you think that it does not look appealing. Try centre aligning the text and move it closer to the bottom rather than keeping it closer to the switch. Maybe it will look better :)

**Comment by rishabh-997** [comment]
> > I tried it and it wasn't looking appealing. I can add another screenshot here with that text vertically alligned in middle
> 
> @rishabh-997 if you think that it does not look appealing. Try centre aligning the text and move it closer to the bottom rather than keeping it closer to the switch. Maybe it will look better :)

I have added screenshot as you said, have a look at all three of them. Let me know your suggestions

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginPresenter.java (hunk 1 above)
Comment (rishabh-997) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginPresenter.java]: The test fails due to some reason which I can't figure out why, is it some error from my side ???

**Comment by f4ww4z** [comment]
@rishabh-997 It looks more appealing with the `get started` at the center. The first screenshot looks good. Also make the 'Unable to Login' text look more like a link / button - I suggest to make it an orange color. Please change the code to reflect this.

**Comment by rishabh-997** [comment]
Screenshot_2020-05-21-22-33-14-91_0b570628ec88a4202b7916ec2df1bebf

**Comment by rishabh-997** [comment]
> @rishabh-997 I would like to suggest a thing like to avoid confusion for users can we give them options for alternate refApp server along with the demo server so that users dont have go on looking for alternate servers URl.

We have now updated the snack bar so when the server is down, there is an apt message regarding **server down error** along with a link with all the alternate servers... Are you suggesting anything different ????

**Comment by LuGO0** [comment]
@rishabh-997 I thought the users getting redirected to some external link will be a bad UX maybe have some sort of drop down to select from a list of servers thats what I thought !!

**Comment by f4ww4z** [comment]
@rishabh-997 Can you fix the merge conflicts and refactor according to the [new localization changes](https://github.com/openmrs/openmrs-contrib-android-client/commit/fba7fe26b3a863862d4a1f4878ac298970d802a3) ? Then we can merge.

File: openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/LoginPresenterTest.java [hunk 2]
```diff
@@ -79,14 +79,22 @@
     private VisitDAO visitDAO;
     @Mock
     private UserService userService;
+<<<<<<< HEAD
+=======
+
+>>>>>>> removed context from presenter
```
Comment (f4ww4z) [inline openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/LoginPresenterTest.java:85]: @rishabh-997 these lines should be removed

File: openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/LoginPresenterTest.java [hunk 3]
```diff
@@ -79,14 +79,22 @@
...
     @Before
     public void setUp() {
         super.setUp();
         VisitRepository visitRepository = new VisitRepository(restApi, visitDAO, locationDAO, new EncounterDAO());
         presenter = new LoginPresenter(restApi, visitRepository, locationDAO, userService, view, openMRS,
+<<<<<<< HEAD
             openMRSLogger, authorizationManager);
+=======
+                openMRSLogger, authorizationManager);
+>>>>>>> removed context from presenter
```
Comment (f4ww4z) [inline openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/LoginPresenterTest.java:97]: Take care of this merge conflict

File: openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/LoginPresenterTest.java [hunk 4]
```diff
@@ -225,12 +233,20 @@ public void shouldLoadLocationsInOnlineMode_allOK() {
     public void shouldLoadLocationsInOnlineMode_errorResponse() {
         mockNetworkConnection(true);
         Mockito.lenient().when(restApi.getLocations(any(), anyString(), anyString()))
+<<<<<<< HEAD
             .thenReturn(mockErrorCall(401));
+=======
+                .thenReturn(mockErrorCall(401));
+>>>>>>> removed context from presenter
```
Comment (f4ww4z) [inline openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/LoginPresenterTest.java:240]: @rishabh-997 see above comment

File: openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/LoginPresenterTest.java [hunk 5]
```diff
@@ -225,12 +233,20 @@ public void shouldLoadLocationsInOnlineMode_allOK() {
...
 
         presenter.loadLocations("someUrl");
         verify(view).initLoginForm(any(), any());
         verify(view).setLocationErrorOccurred(true);
+<<<<<<< HEAD
         verify(view).showInvalidURLSnackbar(R.string.failed_fetching_servers_location);
+=======
+        verify(view).showInvalidURLSnackbar(R.string.snackbar_server_error);
+>>>>>>> removed context from presenter
```
Comment (f4ww4z) [inline openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/LoginPresenterTest.java:249]: @rishabh-997 See above comment

**Pull request merged**